Quick Healthy Picks

ALiyASSER0 opened the pull request:
add use case to get quick healthy food picks sorted by low fat, saturated fat, and carbs

AbdElrahman-Rafaat-Amer committed:
- parse csv file.
- store parsed files in repo.
- create CsvFoodRepository

Took 7 hours 59 minutes

MahmoodTarek committed:
Add GetRandomPotatoMealsUseCase.kt class with functionality to filter meals containing potatoes and return random 10 meals.

MahmoodTarek committed:
Add GetRandomPotatoMealsUseCase.kt class with functionality to filter meals containing potatoes and return random 10 meals.

AbdElrahman-Rafaat-Amer committed:
- fix parseListField

Took 48 minutes

AbdElrahman-Rafaat-Amer committed:
- fix parseNutrition

Took 3 minutes

AbdElrahman-Rafaat-Amer committed:
- create NutritionIndex
- create ColumnIndex
- enhance the variable names

Took 8 minutes

melhanafy-otas committed:
add easy food suggestion feature with corresponding use case and ui component

melhanafy-otas committed:
add guess game feature with ui and use case implementations

ALiyASSER0 committed:
add use case to get quick healthy food picks sorted by low fat, saturated fat, and carbs

AbdElrahman-Rafaat-Amer committed:
- handle the case of file is directory.
- use FileNotFoundException instead of IOException

Took 20 minutes

AbdElrahman-Rafaat-Amer committed:
- use lazily initializing

Took 1 minute

DinaOthman21 committed:
add GetIraqiMealsUseCase and IraqiMealsUi

AbdElrahman-Rafaat-Amer commented on src/main/kotlin/Main.kt:
Don't write the code direct in Main,
U can make it in spreaded function or class related to ur Case UI

AbdElrahman-Rafaat-Amer commented on src/main/kotlin/domain/usecase/GetQuickHealthyPicks.kt:
I think this is the UseCase plz Rename it to have UseCase at the end of the current name to be
GetQuickHealthyPicksUseCase

AbdElrahman-Rafaat-Amer commented on src/main/kotlin/domain/usecase/GetQuickHealthyPicks.kt:
```diff
@@ -0,0 +1,17 @@
+            .filter(::isValidQuickMeal)
+            .sortedBy{ it.nutrition.totalFat + it.nutrition.saturatedFat + it.nutrition.carbohydrates }
+    }
+    private fun isValidQuickMeal(food: Food):Boolean{
```
The task requires that the meal be fast with very low total fat, saturated fat, and carbohydrate values compared to other meals in the dataset.

U didn't check the the other condition for `low total fat, saturated fat, and carbohydrate values `

AbdElrahman-Rafaat-Amer reviewed (commented):
U need to update Ur code to match the task desc

AbdElrahman-Rafaat-Amer committed:
- create GetSeafoodMealsUseCase.kt
- create GetSeafoodMealsUI.kt
- show all seafood meals sorted by protein content

Took 50 minutes

DinaOthman21 committed:
add GetMealsByCountryUseCase and GetMealsByCountryUi

melhanafy-otas committed:
refactor use cases to handle errors and rename classes for consistency

melhanafy-otas committed:
refactor use cases and ui components to improve modularity and naming consistency

DinaOthman21 committed:
Update src/main/kotlin/domain/usecase/GetIraqiMealsUseCase.kt

Co-authored-by: Mohamed Elhanafy <amohamed8684@gmail.com>

DinaOthman21 committed:
Update GetIraqiMealsUseCase.kt

DinaOthman21 committed:
Update GetIraqiMealsUi.kt

AbdElrahman-Rafaat-Amer committed:
- resolve conflicts

Took 3 minutes

melhanafy-otas committed:
add egg-free sweets suggestion feature with corresponding use case and UI component

MahmoodTarek committed:
- Create FoodException.kt to define custom domain exceptions.
- Add GetRandomPotatoMealsUseCase.kt to provide logic for retrieving random meals that contain potatoes.
- Register use case in UseCasesModule.kt for dependency injection.
- Create RandomPotatoMealsUi.kt to handle presentation and display of potato meals.

MahmoodTarek committed:
- Create FoodException.kt to define custom domain exceptions.
- Add GetRandomPotatoMealsUseCase.kt to provide logic for retrieving random meals that contain potatoes.
- Register use case in UseCasesModule.kt for dependency injection.
- Create RandomPotatoMealsUi.kt to handle presentation and display of potato meals.

DinaOthman21 committed:
solve conflict

MahmoodTarek committed:
enhance GetRandomPotatoMealsUseCase.kt and resolve the issue

MahmoodTarek committed:
enhance GetRandomPotatoMealsUseCase.kt and resolve the issue

DinaOthman21 committed:
Update GetMealsByCountryUseCase.kt

DinaOthman21 committed:
Update GetMealsByCountryUi.kt

DinaOthman21 committed:
Update GetMealsByCountryUseCase.kt

melhanafy-otas committed:
refactor FoodException classes to accept dynamic messages and update related use cases

AbdElrahman-Rafaat-Amer committed:
- fix missing imports.
- remove logs parsing.

Took 8 minutes

AbdElrahman-Rafaat-Amer committed:
- fix missing imports.
- remove logs parsing.

Took 13 seconds

AbdElrahman-Rafaat-Amer committed:
- fix false package name

Took 1 minute

ALiyASSER0 committed:
update naming conventions in GetQuickHealthyPicksUseCase

ALiyASSER0 committed:
I renamed the class to GetQuickHealthyPicksUI in the presentation layer.

ALiyASSER0 committed:
resolve conflicts

ALiyASSER0 committed:
delete
